shoop_tests/campaigns/test_context_conditions.py

```diff
@@ -0,0 +1,89 @@
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2016, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+import pytest
+
+from shoop.campaigns.models.campaigns import CatalogCampaign
+from shoop.campaigns.models.context_conditions import (
+    ContactCondition, ContactGroupCondition,
+)
+from shoop.testing.factories import (
+    create_product, create_random_person, get_default_customer_group,
+    get_shop
+)
+from shoop.testing.utils import apply_request_middleware
+
+
+def get_request_for_contact_tests(rf):
+    request = rf.get("/")
+    request.shop = get_shop(prices_include_tax=True)
+    apply_request_middleware(request)
+    return request
+
+
+def create_random_product_and_campaign(shop, rules, original_price_value, discount_value):
+    campaign = CatalogCampaign.objects.create(
+        shop=shop, name="test", discount_amount_value=discount_value, active=True)
+
+    for rule in rules:
+        campaign.conditions.add(rule)
```

Comment:
Why are calling conditions rules? Rule is something that specifies what to do when some conditions match, i.e. campaign is a the rule here.